includes/payment-methods/class-wc-gateway-stripe-alipay.php

```diff
@@ -206,7 +206,7 @@ public function payment_fields() {
 			data-currency="' . esc_attr( strtolower( get_woocommerce_currency() ) ) . '">';
 
 		if ( $description ) {
-			echo wpautop( esc_html( apply_filters( 'wc_stripe_description', wp_kses_post( $description ), $this->id ) ) );
+			echo esc_html( wpautop( apply_filters( 'wc_stripe_description', wp_kses_post( $description ), $this->id ) ) );
```

Comment:
Didn't test this but will this work? I believe `wpautop` adds paragraph tags so if we then run `esc_html` on that, I believe those tags will be escaped and won't render as normal HTML

Reply:
And looks like we're doing this in multiple places so if this doesn't work, we'll need to update all instances

Reply:
You are correct. To properly escape the output while still allowing HTML, we should use `wp_kses_post` after wpautop, it will ensure that only safe HTML tags are allowed.